src/evict/evict_lru.c

```diff
@@ -2145,11 +2145,11 @@ __evict_try_queue_page(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, WT_REF *
      * somehow leave a page without a read generation.
      */
     if (__wt_atomic_load64(&page->read_gen) == WT_READGEN_NOTSET)
-        __wt_evict_read_gen_new(session, page);
+        __wti_evict_read_gen_new(session, page);
 
     /* Pages being forcibly evicted go on the urgent queue. */
     if (modified &&
-      (__wt_atomic_load64(&page->read_gen) == WT_READGEN_OLDEST ||
+      (__wti_evict_page_soon_flagged(&page->read_gen) ||
```

Comment:
Thanks @keitharnoldsmith, I agree, I have now mentioned it [here](https://github.com/wiredtiger/wiredtiger/pull/11087#issuecomment-2398964139) under the `What, if anything, are you concerned about that you'd like the reviewer to focus on?
` section.
 I had a tough time dealing with the different read generations, since this is moving towards a functional change. I am happy to revert the [1fc5939](https://github.com/wiredtiger/wiredtiger/pull/11087/commits/1fc5939dc3d87cec047b84688ea46854f36cee3a) commit for now and only keep the refactoring changes here. This means we would have a getter of read_gen outside the eviction code [here](https://github.com/wiredtiger/wiredtiger/pull/11087/commits/1fc5939dc3d87cec047b84688ea46854f36cee3a#diff-9691a0692cb4c26132c4c27aa260cc6441c2ace4b04487d51769fe84869c624aR172). Let me know you thoughts.

Reply:
We've reverted 1fc5939 (discussion [here](https://github.com/wiredtiger/wiredtiger/pull/11087#discussion_r1798620656)). I've verified this behaviour is back to only queuing urgent pages and not `WONT_NEED` pages.